api/api.go

```diff
@@ -1466,28 +1474,36 @@ func (api *Server) reverseActionsInBlock(blk *block.Block, reverseStart, count u
 				Action:    selp.Proto(),
 				ActHash:   hex.EncodeToString(actHash[:]),
 				BlkHash:   blkHash,
+				Timestamp: ts,
 				BlkHeight: blkHeight,
 				Sender:    sender.String(),
-				Timestamp: ts,
+				Index:     uint32(ri),
 			},
 		}, res...)
 	}
 	return res
 }
 
 func (api *Server) getLogsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
+	h, err := api.dao.GetBlockHash(blockNumber)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
 	logBloomFilter, err := api.bfIndexer.BlockFilterByHeight(blockNumber)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
+
 	if !filter.ExistInBloomFilterv2(logBloomFilter) {
 		return nil, nil
 	}
 	receipts, err := api.dao.GetReceipts(blockNumber)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	return filter.MatchLogs(receipts), nil
+
```

Comment:
i think move `h, err := api.dao.GetBlockHash(blockNumber)` here, right before using it

in case error return above, no need to call this